bootstrap/api/v1beta2/ck8sconfig_types.go

```diff
@@ -66,6 +66,18 @@ type CK8sConfigSpec struct {
 	// +optional
 	SnapstoreProxyID string `json:"snapstoreProxyId,omitempty"`
 
+	// HTTPSProxy is optional https proxy configuration
+	// +optional
+	HTTPSProxy string `json:"httpsProxy,omitempty"`
+
+	// HTTPProxy is optional http proxy configuration
+	// +optional
+	HTTPProxy string `json:"httpProxy,omitempty"`
+
+	// NoProxy is optional no proxy configuration
+	// +optional
+	NoProxy string `json:"noProxy,omitempty"`
```

Comment:
I'm not too happy with the naming here as these are only getting set for containerd. We should make it explicit like `ContainerdHTTPProxy`  if this is what we want.